### ISSUE TITLE
aws_launch_template: add the option to tag network-interfaces

### ISSUE THREAD

**Issue by AlexisJackson** [hashicorp/terraform-provider-aws#19140]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The LaunchTemplateTagSpecificationRequest API docs show that network-interface is an option. Can this be added to the list of supported types? We pair ECS Clusters with AutoScaling Groups and it would be helpful if the network-interfaces/enis attached to the EC2 Instances were tagged along with the instances and volumes.

https://github.com/hashicorp/terraform-provider-aws/blob/main/aws/resource_aws_launch_template.go#L588-L591

### New or Affected Resource(s)

* aws_launch_template

### Potential Terraform Configuration

```hcl
tag_specifications {
    resource_type = "network-interface"

    tags = {
      Name = "test"
    }
}
```

### References

* [AWS LaunchTemplateTagSpecificationRequest API docs](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_LaunchTemplateTagSpecificationRequest.html)

**Comment by Aym3nTN** [comment]
Really appreciate adding this feature, we're building several EKS clusters with severals nodes for our different environments and we need to efficiently track our spending.

**Comment by alexmnyc** [comment]
same here. it is a problem! why not allow the entire list of tag specification allowed enum?

**Comment by gordonbondon** [comment]
Should be fixed by #20409
